src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,205 @@
+                    output = taskList.addEvent(event);
+                }
+            } else {
+                throw new InvalidInputException();
+            }
+        return output;
+        }
+
+    /**
+     * Process tasks from the database to a list.
+     * @param line Every line of task from the database.
+     * @param storedTasks The list of task to load with.
+     */
+    public void processInputFromFile(String line, List<Task> storedTasks) {
+        String[] split = line.split("\\ \\|\\ ");
+        String type = split[0];
+        switch (type) {
+            case "T":
+                Task task = new Task(split[2]);
+                if (Integer.parseInt(split[1]) == 1) {
+                    task.setDone();
+                }
+                storedTasks.add(task);
+                break;
+            case "E":
+                String eventTime = split[3];
+                String eventDescription = split[2];
+                int eventStatus = Integer.parseInt(split[1]);
+                if (isValidDate(eventTime)) {
+                    LocalDate eld = LocalDate.parse(eventTime);
+                    Event event = new Event(eventDescription, eld);
+                    if (eventStatus == 1) {
+                        event.setDone();
+                    }
+                    storedTasks.add(event);
+                } else {
+                    Event event = new Event(eventDescription, eventTime);
+                    if (eventStatus == 1) {
+                        event.setDone();
+                    }
+                    storedTasks.add(event);
+                }
+                break;
+            case "D":
+                String deadlineTime = split[3];
+                String deadlineDescription = split[2];
+                int deadlineStatus = Integer.parseInt(split[1]);
+                if (isValidDate(deadlineTime)) {
+                    LocalDate dld = LocalDate.parse(deadlineTime);
+                    Deadline deadline = new Deadline(deadlineDescription, dld);
+                    if (deadlineStatus == 1) {
+                        deadline.setDone();
+                    }
+                    storedTasks.add(deadline);
+                } else {
+                    Deadline deadline = new Deadline(deadlineDescription, deadlineTime);
+                    if (deadlineStatus == 1) {
+                        deadline.setDone();
+                    }
+                    storedTasks.add(deadline);
+                }
+                break;
+            default:
+                break;
+        }
+    }
+
+    /**
+     * A logic function to determine the validity of a date entered.
+     * A date is a string in special format entered by the user to create deadlines/events.\
+     * with the format YYYY-MM-DD
+     * @param str The string of date to be tested with.
+     * @return A boolean to denote this string is a date.
+     */
+    static boolean isValidDate(String str) {
```

Comment:
Could the exception be caught when constructing the Deadline or Event entity? Then this method would not be needed as an additional step.